packages/hardhat-plugin/src/index.ts

```diff
@@ -247,7 +248,15 @@ ignitionScope
       id
     );
 
-    const statusResult = await status(deploymentDir);
+    let statusResult: StatusResult;
+    try {
+      statusResult = await status(deploymentDir);
+    } catch (e) {
+      throw new NomicLabsHardhatPluginError(
```

Comment:
I think it'd be better to selectively rethrow, something like this:


```
catch (e) {
  if (e instanceOf IgnitionError && shouldBeHardhatPluginError(e)) {
    throw new ...
  }
  
  throw e;
}
```

```
function shouldBeHardhatPluginError(error:IgnitionError):boolean {
   /// white list of errors by their code
}
```

Reply:
that way we can use that logic on all the tasks, and make sure we are handling all the errors we care about

Reply:
done!